pandas/tests/io/parser/common/test_ints.py

```diff
@@ -110,6 +116,85 @@ def test_integer_overflow_bug(all_parsers, sep):
     tm.assert_frame_equal(result, expected)
 
 
+def _iinfo(dtype):
+    pdtype = pandas_dtype(dtype)
+    iinfo = np.iinfo(pdtype.type if is_extension_array_dtype(dtype) else pdtype)
+    return iinfo
+
+
+_raises_any_integer_cast_exception = pytest.raises(  # noqa: PDF010
+    (OverflowError, TypeError, ValueError),
+    match="|".join(
+        [
+            "Overflow",
+            "cannot safely cast non-equivalent",
+            "Integer out of range",
+            "Unable to convert column",
+            "The elements provided in the data cannot all be casted to the dtype",
+        ]
+    ),
+)
+
+
+@skip_pyarrow
+@pytest.mark.parametrize(
+    "getval,expected",
+    [
+        (lambda dtype: _iinfo(dtype).max, nullcontext()),  # in range does not raise
+        (lambda dtype: _iinfo(dtype).min, nullcontext()),  # in range does not raise
+        (lambda dtype: _iinfo(dtype).max + 1, _raises_any_integer_cast_exception),
+        (lambda dtype: _iinfo(dtype).min - 1, _raises_any_integer_cast_exception),
+    ],
+)
+def test_integer_overflow_with_user_dtype(all_parsers, any_int_dtype, getval, expected):
+    dtype = any_int_dtype
+    parser = all_parsers
+    val = getval(dtype)
+    data = f"A\n{val}"
+
+    # Positive value overflow with uint8, uint16, uint32 and any overflow with
+    # int8, int16, int32 only throw a FutureWarning until deprecation from #41734
+    # becomes enforced. After enforcement, the following block must be deleted.
+    if (
+        (expected == _raises_any_integer_cast_exception)
+        and (parser.engine == "python")
+        and (not is_extension_array_dtype(dtype))
+        and (dtype < np.dtype("int64"))
+        and not (is_unsigned_integer_dtype(dtype) and (val < 0))
+    ):
+        expected = tm.assert_produces_warning(
+            FutureWarning,
+            match=f"Values are too large to be losslessly cast to {np.dtype(dtype)}.",
+            check_stacklevel=False,
+        )
+
+    with expected:
+        result = parser.read_csv(StringIO(data), dtype=dtype)
+    if isinstance(expected, nullcontext):
+        expected_result = DataFrame({"A": [val]}, dtype=dtype)
+        tm.assert_frame_equal(result, expected_result)
+
+
+@skip_pyarrow
+@pytest.mark.parametrize(
+    "val,expected",
+    [
+        (0.0, nullcontext()),  # lossless conversion does not raise
+        (0.1, pytest.raises((TypeError, ValueError), match=None)),  # noqa: PDF010
+    ],
+)
+def test_integer_from_float_raises(all_parsers, any_int_dtype, val, expected):
```

Comment:
Just make two tests for these things

Reply:
Okay, see [39b5c91](https://github.com/pandas-dev/pandas/pull/47168/commits/39b5c91171bd5da6876f95c077c74c4597f645b1)